templates/IAM/iam-role.yaml

```diff
@@ -0,0 +1,41 @@
+# Define a role
+#
+AWSTemplateFormatVersion: '2010-09-09'
+Description: 'Creates a role'
+Parameters:
+  AssumeRolePolicyDocument:
+    Type: String
+    Description: Policy describing trust relationships
+    Default: ''
```

Comment:
if you make this optional then you need to handle it with a condition, otherwise if the user doesn't pass in something for this the deployment will fail.